tests/sources/xml/test_ead.py

```diff
@@ -515,19 +576,223 @@ def test_ead_record_correct_identifiers_from_multiple_unitid(caplog):
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <archdesc level="collection">
+                <did>
+                    <unittitle>
+                        Charles J. Connick Stained Glass
+                        <emph>
+                            Foundation
+                            <emph>Collection</emph>
+                        </emph>
+                        <num>VC.0002</num>
+                    </unittitle>
+                    <unittitle></unittitle>
+                    <unittitle></unittitle>
+                </did>
+            </archdesc>
+            """
+        )
+    )
+    assert Ead.get_alternate_titles(source_record) is None
+
+
+def test_get_alternate_titles_transforms_correctly_if_fields_missing():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <archdesc level="collection">
+                <did>
+                    <unittitle>
+                        Charles J. Connick Stained Glass
+                        <emph>
+                            Foundation
+                            <emph>Collection</emph>
+                        </emph>
+                        <num>VC.0002</num>
+                    </unittitle>
+                </did>
+            </archdesc>
+            """
+        )
+    )
+    assert Ead.get_alternate_titles(source_record) is None
+
+
+def test_get_citation_success():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <archdesc level="collection">
+                <prefercite>
+                    <head>Preferred Citation</head>
+                    <p>Charles J. Connick Stained Glass Foundation Collection, VC-0002, box X. Massachusetts Institute of Technology, Department of Distinctive Collections, Cambridge, Massachusetts.</p>
+                </prefercite>
+            </archdesc>
+            """
+        )
+    )
+    assert Ead.get_citation(source_record) == (
+        "Charles J. Connick Stained Glass Foundation Collection, VC-0002, box X. Massachusetts Institute of Technology, Department of Distinctive Collections, Cambridge, Massachusetts."
+    )
+
+
+def test_get_citation_transforms_correctly_if_fields_blank():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <archdesc level="collection">
+                <prefercite></prefercite>
+            </archdesc>
+            """
+        )
+    )
+    assert Ead.get_citation(source_record) is None
+
+
+def test_get_citation_transforms_correctly_if_fields_missing():
+    source_record = create_ead_source_record_stub(
+        metadata_insert=(
+            """
+            <archdesc level="collection"></archdesc>
+            """
+        )
+    )
+    assert Ead.get_citation(source_record) is None
+
+
+def test_get_main_titles_success():
```

Comment:
This would be easier to read if the order matched the method order in `ead.py` so I would move the `get_main_titles` tests further down

Reply:
Updated!